src/slash.c

```diff
@@ -34,7 +34,7 @@ unsigned long long slash(const unsigned char* key, int length) {
 		case 4: combine(result, (unsigned long long)data[3] << 24);
 		case 3: combine(result, (unsigned long long)data[2] << 16);
 		case 2: combine(result, (unsigned long long)data[1] << 8);
-		case 1: combine(result, (unsigned long long)data[0]);
```

Comment:
This change can cause a potential segfault. The loop above increments the `data` pointer and decrements the `length` — this implies that the length is relative to the new `data`. If the length is `0` and it goes into the `default` case, it will attempt to read 8 bytes from the current `data` array but will get garbage or segfault.

It should stay at `1` so that if one byte remains after reading blocks of 8 bytes it can be combined into the result for further mixing.